File: solution/001TwoSum/001TwoSum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for j := 0; j < len(nums); j++ {
		if i, ok := m[nums[j]]; !ok {
			m[target-nums[j]] = j
		} else {
			return []int{i, j}
		}
	}
	return nil
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Printf("%v\n", twoSum(nums, target))
}

File: solution/001TwoSum/001TwoSum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut m = std::collections::HashMap::new();
        for i in 0..nums.len() {
            if let Some(j) = m.get(&nums[i]) {
                return vec![*j as i32,i as i32];
            }else{
                m.insert(target-nums[i], i);
            }
        }
        vec![]
    }
}
